x/staging/posts/simulation/operations_posts.go

```diff
@@ -91,10 +92,19 @@ func sendMsgCreatePost(
 
 // randomPostCreateFields returns the creator of the post as well as the parent id
 func randomPostCreateFields(
-	r *rand.Rand, ctx sdk.Context, accs []simtypes.Account, k keeper.Keeper, ak authkeeper.AccountKeeper,
+	r *rand.Rand, ctx sdk.Context, accs []simtypes.Account, k keeper.Keeper, ak authkeeper.AccountKeeper, sk subspaceskeeper.Keeper,
 ) (*PostData, bool) {
 
 	postData := RandomPostData(r, accs)
+	subspaceData := RandomSubspace(r, accs)
+
+	if err := sk.SaveSubspace(ctx, subspaceData.Subspace, subspaceData.Subspace.Creator); err != nil {
+		return nil, true
+	}
```

Comment:
We should change how the subspace is retrieved here and in all the other places of the simulation operations as well